### PR TITLE
Update shadertype.md

### PR DESCRIPTION
**Pull request by TylerDahl**
**What does this PR do?**
Update the shadertype documentation to remove auto-generated text and add a description and example.

Thanks for the contribution! Please provide a concise description of the problem this request solves.

**How does this PR change Premake's behavior?**
Modifies the Premake documentation.

Are there any breaking changes? Will any existing behavior change?
No.

**Anything else we should know?**
No.

Add any other context about your changes here.

**Did you check all the boxes?**

- [CHECK] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [CHECK ] Add unit tests showing fix or feature works; all tests pass
- [CHECK ] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [CHECK ] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [CHECK ] Minimize the number of commits
- [CHECK ] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

**Review by KyrietS — APPROVED** [review]
You could add some more info about this parameter, but I guess It's better than nothing. Thanks.
Note that we have other shaderX pages that are auto-generated. If you are somwehat involved in compiling shaders with Premake then please don't hesitate to improve other pages as well 🙃

File: website/docs/shadertype.md
Comment (nickclark2016) [inline website/docs/shadertype.md:4]: Let's use `shadertype ("value")` instead. It's more consistent with other docs.

**Review by nickclark2016 — CHANGES_REQUESTED** [review]

File: website/docs/shadertype.md
Comment (KyrietS) [reply website/docs/shadertype.md:4]: +1

**Comment by nickclark2016** [comment]
Sorry, one last minor thing and I'll leave you alone. Instead of making a new commit, could you just amend the previous?

```sh
git reset HEAD~
git add --all
git commit --amend --no-edit
```

We try to keep the number of commits to a minimum (this should be applied to all your PRs).

**Comment by KyrietS** [comment]
@TylerDahl please squash your commits if you want to be listed on git blame (and our website) as an author of this change.

1. Go to your repo and checkout your branch `patch-1`: `git checkout patch-1`
2. `git reset --soft master`
3. `git add .`
4. `git commit -m "Update shadertype.md"`
5. `git push --force`

If you don't respond then I'll merge your change using squash method. 
Also @nickclark2016 is your *change request* resolved? Do you approve this commit in it's current state?

**Comment by TylerDahl** [comment]
@KyrietS Just squashed my commits.

**Comment by KyrietS** [comment]
Let's wait for @nickclark2016's approval

**Review by samsinsane — APPROVED** [review]

**Pull request merged**
